packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -129,6 +130,26 @@ export class ContextualMenuBase extends React.Component<IContextualMenuProps, IC
   // tslint:disable-next-line:deprecation
   private _classNames: IProcessedStyleSet<IContextualMenuStyles> | IContextualMenuClassNames;
 
+  private _convertMenuItemToMenuItemRenderProps = memoizeFunction(
```

Comment:
There are some benefits and draw backs to adding this function.

Pro: 
We take advantage of shouldComponentUpdate for menu items and reduce menu item rerenders from ms time to microsecond time. For example this will reduce subsequent render times for Office Online apps by 10-15% based on test profiling I was running with and without the change.

con: 
We will use up more memory as we have to build the cache for the props and since we are using an object in the memoization this can grow the cache to its limit for each menu if the menu item changes on each opening or there isnt proper memoization happening on the implementation side.
We can limit this issue by reducing the cache limit for this function or by making this an external function so we will only hit the cache limit once.

Reply:
i think in case like this, what we need to do is to make sure we don't mutate the item props user passed instead doing memoization. if user passes mutating props (which can happen very easily :(), we don't end up filling up the cache

see my comment here: https://github.com/microsoft/fluentui/pull/12831#pullrequestreview-398729883

Reply:
Thank you for the detailed response! I've made the changes proposed and have verified it addresses the issue.